### PR TITLE
Pass finder now accepts combined test suite. 

### PR DESCRIPTION
**Pull request by nmdis1999**
This PR addressed:
1) Changes to `lib.rs` so that `mk_compiler` accepts current supported `prelink-pipeline` and `postlink-pipeline` flags and their passes. 
2) Changed `yklua_test.sh` to now run both YK and YKLUA tests. 
3) Adjusted `try_passes.py` script to run test pipeline on the new combined test suite script.

### CODE REVIEW

File: tests/src/lib.rs [hunk 1]
```diff
@@ -119,14 +119,35 @@ pub fn mk_compiler(
     #[cfg(cargo_profile = "release")]
     let mode = "release";
 
+    let prelink_args = env::var("PRELINK_PASSES").unwrap_or_default();
+    let postlink_args = env::var("POSTLINK_PASSES").unwrap_or_default(); 
+
+    let mut yk_config_args: Vec<&str> = Vec::new();
```
Comment (ltratt) [inline tests/src/lib.rs:125]: This and the next line can probably become `let mut yk_config_args = vec![mode]`.

File: tests/src/lib.rs [hunk 2]
```diff
@@ -119,14 +119,35 @@ pub fn mk_compiler(
...
+    yk_config_args.push(mode);
+
+    if !prelink_args.is_empty() {
+        yk_config_args.push("--prelink-pipeline");
+        yk_config_args.push(&prelink_args);
+    }
+    yk_config_args.push("--cflags");
+    yk_config_args.push("--cppflags");
+    if !postlink_args.is_empty() {
+        yk_config_args.push("postlink-pipeline");
```
Comment (ltratt) [inline tests/src/lib.rs:135]: Is this supposed to be `--postlink-pipeline`?

File: tests/src/lib.rs [hunk 3]
```diff
@@ -119,14 +119,35 @@ pub fn mk_compiler(
...
+    let postlink_args = env::var("POSTLINK_PASSES").unwrap_or_default(); 
+
+    let mut yk_config_args: Vec<&str> = Vec::new();
+    yk_config_args.push(mode);
+
+    if !prelink_args.is_empty() {
+        yk_config_args.push("--prelink-pipeline");
+        yk_config_args.push(&prelink_args);
+    }
+    yk_config_args.push("--cflags");
```
Comment (ltratt) [inline tests/src/lib.rs:132]: I think lines 132, 133, 138, and 139 can be part of the initial vec creation? I don't think their order is important?

File: tests/src/lib.rs [hunk 4]
```diff
@@ -119,14 +119,35 @@ pub fn mk_compiler(
...
+    }
+    yk_config_args.push("--ldflags");
+    yk_config_args.push("--libs");
+    
     let yk_config_out = Command::new(yk_config)
-        .args([mode, "--cflags", "--cppflags", "--ldflags", "--libs"])
+        .args(&yk_config_args)
         .output()
         .expect("failed to execute yk-config");
+
```
Comment (ltratt) [inline tests/src/lib.rs:145]: I don't think this added blank line helps (or line 150).

File: ykrt/pass_finder/run_tests.sh [hunk 5]
```diff
@@ -1,7 +1,24 @@
 #!/bin/sh
+# This script runs tests cases for YK and YKLUA. The aim is
```
Comment (ltratt) [inline ykrt/pass_finder/run_tests.sh:2]: We generally prefer formatting comments to 80 or 100 chars rather than 60.

File: ykrt/pass_finder/run_tests.sh [hunk 6]
```diff
@@ -1,7 +1,24 @@
...
+# to feed this test script to try_passes.py and get list of
+# passes which are successful for all test suites. We can expand 
+# this script to accomodate new test suites in future.
+
+# how do I pass mode for yk through terminal?
```
Comment (ltratt) [inline ykrt/pass_finder/run_tests.sh:7]: Not sure I understand the question. Maybe @vext01 does?

File: tests/src/lib.rs (hunk 3 above)
Comment (nmdis1999) [reply tests/src/lib.rs:132]: so `--prelink-pipeline` and `--postlink-pipeline` should be passed before  `--cflags` and `--ldflags` respectively.

File: tests/src/lib.rs (hunk 2 above)
Comment (nmdis1999) [reply tests/src/lib.rs:135]: yes, will fix this.

File: ykrt/pass_finder/run_tests.sh (hunk 6 above)
Comment (vext01) [reply ykrt/pass_finder/run_tests.sh:7]: I don't understand either :)

Please can you rephrase the question?

Comment (nmdis1999) [reply ykrt/pass_finder/run_tests.sh:7]: @vext01 with current script, we just enter YK repository and run `cargo test`. I was wondering, while testing should the default mode be `Release` (because your original test script for YKLUA sets the mode as `Release`)? If yes, then I don't know how to set the mode to `Release` for YK.

Comment (nmdis1999) [reply ykrt/pass_finder/run_tests.sh:7]: yes, thank you!

File: tests/src/lib.rs (hunk 3 above)
Comment (ltratt) [reply tests/src/lib.rs:132]: OK.

File: tests/src/lib.rs (hunk 1 above)
Comment (nmdis1999) [reply tests/src/lib.rs]: addressed here: 9e4bf0f2470deb6e2fa8cc2fd5aa2fe12d225bff

File: tests/src/lib.rs (hunk 2 above)
Comment (nmdis1999) [reply tests/src/lib.rs]: adressed here: 9e4bf0f2470deb6e2fa8cc2fd5aa2fe12d225bff

File: ykrt/pass_finder/run_tests.sh (hunk 5 above)
Comment (nmdis1999) [reply ykrt/pass_finder/run_tests.sh]: addressed here: 147f01131d3ccab617b903709b6fdf561a9928f0

**Comment by vext01** [comment]
Is this ready for re-review?

**Comment by nmdis1999** [comment]
> Is this ready for re-review?

yes!

**Comment by vext01** [comment]
OK. In future, make sure you let your reviewers know!

File: ykrt/pass_finder/run_tests.sh (hunk 5 above)
Comment (vext01) [reply ykrt/pass_finder/run_tests.sh]: Did you link the right commit there?

Comment (nmdis1999) [reply ykrt/pass_finder/run_tests.sh]: sorry, updated!

**Comment by vext01** [comment]
Please squash and sync the branch.

**Comment by nmdis1999** [comment]
@vext01 done!

**Comment by vext01** [comment]
Please push a commit to fix formatting errors (run `cargo fmt`).

**Comment by vext01** [comment]
Remember the rule: don't squash unless the reviewer asks.

The problem is: I can't verify what changed with the new change squashed in. Next time, push just the commit, so I can look at it first, then if all is well I will ask you to squash again.

bors r+
